plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -88,7 +88,7 @@ public function get_formatted_log( $level, $message, $context = array() ) {
 			unset( $context['tags'] );
 		}
 
-		if ( class_exists( '\WC_Tracks' ) ) {
+		if ( class_exists( '\WC_Tracks' ) && function_exists( 'wp_get_current_user' ) ) {
```

Comment:
We might be better to wrap the whole thing in try/catch 🤔  I suppose there could be all sorts of possible errors associated with calling other functions when the site is in a fatal error state. Rather than just protecting against this this function not existing, we could try catch a larger surface.

Reply:
There's a try/catch just one level above though, https://github.com/woocommerce/woocommerce/blob/573d1bebed2543f2f34e37dcb7abf01ebcf32522/plugins/woocommerce/src/Internal/Logging/RemoteLogger.php#L201

I think we need to change something about that try-catch as mentioned in a separate chat

Reply:
Right, sorry I forgot try-catch will still let fatal errors slip through.

I could see it being beneficial to create some guard rails around every non-inbuilt function -- ` wp_parse_url`, `WC`, `wp_get_current_user` etc -- by making private method wrappers/aliases. Something like:

```
private function wp_parse_url() {
 if ( function_exists( 'wp_parse_url' ) ) {
   wp_parse_url();
 }
 else {
   return null; // or alternatively throw an exception 🤔 
 }
}
```

Reply:
> What do you think if we require pluggable.php explicitly when wp_get_current_user is not defined? 

After thinking about it a bit, I think this might not be necessary if we will likely change `WC_Tracks` to use the site language.